### ISSUE TITLE
Compatibilité avec TywellControl

### ISSUE THREAD

**Issue by SantoOfEmerald** [fmartinou/tydom2mqtt#191]
Bonjour,

Je commence par vous remercier pour votre travail. J'ai presque tous les capteurs et actionneurs qui remontent, même ceux du pack Tysense.

Les deux seuls que je ne retrouve pas sont ceux du tywell control, température et hydrométrie intérieur.
![tywell control](https://github.com/fmartinou/tydom2mqtt/assets/168841777/e2a20b25-06b2-4585-a89c-f8f5584b6fa4)

j'ai ce message dans les logs : 
2024-05-31 10:37:14,771 - sensors.Sensor       - INFO    - Sensor created / updated : lightPower 455
2024-05-31 10:40:15,872 - tydom.MessageHandler - INFO    - Device update (id=1709900987, endpoint=1709900987, name=Tysense Thermo 1, type=sensorThermo)
2024-05-31 10:40:15,872 - sensors.Sensor       - INFO    - Sensor created / updated : outTemperature 16.61
2024-05-31 10:41:07,439 - tydom.MessageHandler - WARNING - Unknown device name (1704828778_1704828778)
2024-05-31 10:41:07,440 - tydom.MessageHandler - WARNING - Unknown device type (1704828778_1704828778)
2024-05-31 10:41:07,440 - tydom.MessageHandler - INFO    - Device update (id=1704828778, endpoint=1704828778, name=, type=)
2024-05-31 10:42:24,029 - tydom.MessageHandler - INFO    - Device update (id=1709900833, endpoint=1709900833, name=Tysense Sun 1, type=sensorSun)
2024-05-31 10:42:24,030 - sensors.Sensor       - INFO    - Sensor created / updated : lightPower 637
2024-05-31 10:44:48,157 - tydom.MessageHandler - WARNING - Unknown device name (1709901215_1709901215)
2024-05-31 10:44:48,158 - tydom.MessageHandler - WARNING - Unknown device type (1709901215_1709901215)
2024-05-31 10:44:48,158 - tydom.MessageHandler - INFO    - Device update (id=1709901215, endpoint=1709901215, name=, type=)

Je vois bien les deux capteurs Tysense ( d=1709900987 sensorThermo et id=1709900833 sensorSun).

Je suppose que les deux id : 1704828778 et 1709901215 doivent être les deux que je cherche.

Savez-vous comment je peux les ajouter ? 

Merci.

**Comment by SantoOfEmerald** [comment]
Bonjour,

pour le moment non, j'ai toujours ces message dans le log du module  :

```
2024-07-19 15:34:55,849 - sensors.Sensor       - INFO    - Sensor created / updated : lightPower 528
2024-07-19 15:39:39,157 - tydom.MessageHandler - WARNING - Unknown device name (1704828778_1704828778)
2024-07-19 15:39:39,157 - tydom.MessageHandler - WARNING - Unknown device type (1704828778_1704828778)
2024-07-19 15:39:39,157 - tydom.MessageHandler - INFO    - Device update (id=1704828778, endpoint=1704828778, name=, type=)
2024-07-19 15:40:03,467 - tydom.MessageHandler - INFO    - Device update (id=1709900833, endpoint=1709900833, name=Tysense Sun 1, type=sensorSun)
```

je n'arrive pas a faire apparètre le device qui à l'id 1704828778 ni as lui définir un type.

**Comment by Twinuts59** [comment]
Hello, petit up car j'ai le même souci. Les Tywell control ne remontent pas :(

**Comment by AlBla84140** [comment]
Pour mon cas, j'ai réussi a récupérer mes Tysense thermo avec cet addon :
https://github.com/koleos6/hassio-addons
Mais il ne remonte pas les lights chez moi.
Donc j'utilise les deux ensemble pour tout remonter.
A vor pour vous.

**Comment by SantoOfEmerald** [comment]
en effet j'utilise cette Addon pour remonté les tysense thermo et sun.

j'ai bien la température extérieur et le retour du capteur de luminosité, mais c'est les retour de température et d'hydrométries intérieur via le tywelcontrol qui pour le moment ne remonte pas.

**Comment by Twinuts59** [comment]
Exact, we are speaking here about Tywell control, that is a lot of deploy in all new house install with delta dore.

**Comment by tdubouchet** [comment]
Hello, petit up, même soucis avec les Tywell Control, pas de température / hydrométrie :(

![image](https://github.com/user-attachments/assets/41c530bb-de97-48b9-8c3d-95154eece948)

**Comment by advitae** [comment]
Hello à tous,

un petit up car idem impossible de récupérer les infos du Tywell Control...

**Comment by SantoOfEmerald** [comment]
Bonjour

@tdubouchet, même si je n'ai pas les informations, je suis curieux de savoir comment vous avez fait pour l'ajouté à une carte ? 

Je ne le vois même pas dans la liste des entités.

**Comment by Charly3944** [comment]
Est ce que certains d'entre vous ont réussi a résoudre ceci?

**Comment by rodtrip2** [comment]
Même problème avec le pack tywell pro.
On ne peut actuellement pas remonter les infos du thermostat tywell control (réglage de la T° de consigne, T° et humidité du thermostat, chauffage en marche ou non).
Par contre, je ne sais pas comment remonter les infos de l'API nécessaires pour ajouter cette fonctionnalité, si qqun peut m'aider à aider. ^^

**Comment by AlBla84140** [comment]
Le problème est de savoir qui est capable d'ajouter ce type (sensors) dans le programme python.
Mes petites connaissances me permettent de comprendre le programme, de tirer quelques informations en analysant les différents fichiers py.
Mais pour ajouter des modules à prendre en compte il faut savoir quelles infos remontent des modules tywell control.
Par exemple ceci pour un light 
`{
  "name": null,
  "brightness_scale": 100,
  "unique_id": "1569531422_1569531422",
  "optimistic": true,
  "brightness_state_topic": "light/tydom/1569531422_1569531422/current_level",
  "brightness_command_topic": "light/tydom/1569531422_1569531422/set_level",
  "command_topic": "light/tydom/1569531422_1569531422/set_levelCmd",
  "state_topic": "light/tydom/1569531422_1569531422/current_level",
  "json_attributes_topic": "light/tydom/1569531422_1569531422/attributes",
  "payload_off": "OFF",
  "payload_on": "ON",
  "on_command_type": "brightness",
  "retain": "false",
  "state_value_template": "{{ 'ON' if value | int(0) > 0 else 'OFF' }}",
  "device": {
    "manufacturer": "Delta Dore",
    "model": "Lumiere",
    "name": "Cellier",
    "identifiers": "1569531422_1569531422"
  }
`

**Comment by patoche94** [comment]
Bonjour, même soucis : pas de remontée des infos du Tywell Control 
Il apparaît bien dans les équipements mais toutes les infos sont vides

**Comment by batouchu** [comment]
As shared in the linked issue above, @CharlesP44 identified the correct device type: `re2020ControlBoiler` (see https://github.com/CyrilP/hass-deltadore-tydom-component/pull/233 ). Not sure how well HASS components translate to tydom2MQTT
